extensions-core/kinesis-indexing-service/src/test/java/org/apache/druid/indexing/kinesis/KinesisIndexTaskTest.java

```diff
@@ -2696,17 +2690,14 @@ public void testIncrementalHandOffReadsThroughEndOffsets() throws Exception
     Assert.assertEquals(checkpoint1.getPartitionSequenceNumberMap(), currentOffsets);
     task.getRunner().setEndOffsets(currentOffsets, false);
 
-    // The task is supposed to consume remaining rows up to the offset of 13
     while (task.getRunner().getStatus() != Status.PAUSED) {
       Thread.sleep(10);
     }
     currentOffsets = ImmutableMap.copyOf(task.getRunner().getCurrentOffsets());
     Assert.assertEquals(checkpoint2.getPartitionSequenceNumberMap(), currentOffsets);
 
-    task.getRunner().setEndOffsets(
-        ImmutableMap.of(shardId1, String.valueOf(Long.valueOf(task.getRunner().getCurrentOffsets().get(shardId1)) + 1)),
-        true
-    );
+    // Set end offsets to one past the checkpoint, simulating a replica that needs to catch up.
+    task.getRunner().setEndOffsets(ImmutableMap.of(shardId1, "10"), true);
```

Comment:
FYI, I fixed this test to be more realistic in https://github.com/apache/incubator-druid/pull/7264.

Reply:
It looks like that PR has enough approvals to commit. I'll do that and merge it into this one.